src/systemtask/SystemTask.cpp

```diff
@@ -483,12 +486,11 @@ void SystemTask::UpdateMotion() {
   motionController.IsSensorOk(motionSensor.IsOk());
   motionController.Update(motionValues.x, motionValues.y, motionValues.z, motionValues.steps);
 
-  if (settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::RaiseWrist) &&
-      motionController.Should_RaiseWake(state == SystemTaskState::Sleeping)) {
-    GoToRunning();
-  }
-  if (settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::Shake) &&
-      motionController.Should_ShakeWake(settingsController.GetShakeThreshold())) {
+  if (settingsController.GetNotificationStatus() != Controllers::Settings::Notification::Sleep &&
```

Comment:
this is a very long convoluted conditional making my head dizzy. I think it would be better for readability and future bug prevention to keep this conditional simpler (even though we'd have to use more lines of code for it)

Reply:
Would it be enough to separate the notification status check, but keep the wakeup methods in one?

Reply:
I think so, yes